packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts

```diff
@@ -0,0 +1,88 @@
+import type { Hole } from '../../types'
+import {
+  isConsecutiveWithPrevious,
+  Direction,
+} from '../../../packages/is-consecutive-with-previous'
+
+export enum HoleErrors {
+  ConflictingDrawBends = 'CONFLICTING_DRAW_BENDS',
+  ConflictingBlowBends = 'CONFLICTING_BLOW_BENDS',
+  TooManyBends = 'TOO_MANY_BENDS',
+  TooManyBlowbends = 'TOO_MANY_BLOWBENDS',
+  TooManyOverblows = 'TOO_MANY_OVERBLOWS',
+  TooManyOverdraws = 'TOO_MANY_OVERDRAWS',
+  NonconsecutiveBends = 'NONCONSECUTIVE_BENDS',
+  NonconsecutiveBlowbends = 'NONCONSECUTIVE_BLOWBENDS',
+  NonconsecutiveOverblows = 'NONCONSECUTIVE_OVERBLOWS',
+  NonconsecutiveOverdraws = 'NONCONSECUTIVE_OVERDRAWS',
+  ConflictingValvedBlowBends = 'CONFLICTING_VALVED_BLOW_BENDS',
+  ConflictingValvedDrawBends = 'CONFLICTING_VALVED_DRAW_BENDS',
+}
+
+export const isHoleValid = (hole: Hole): HoleErrors[] => {
+  const bendLimit = 5
```

Comment:
Add a comment here that this limit is actually more of a comment on the sensible limit of reed  distance. Bends won't really be possible at all if the reeds are far enough apart to be attempting to produce 6 semitone bends.

Perhaps this limit should be discovered more directly by just counting the distance between the reeds. That's more direct, but is it as semantic? I think not.

Reply:
Done